Source/Core/Core/HW/EXI/EXI_DeviceEthernet.h

```diff
@@ -297,34 +297,97 @@ class CEXIETHERNET : public IEXIDevice
   std::unique_ptr<u8[]> mBbaMem;
   std::unique_ptr<u8[]> tx_fifo;
 
-  // TAP interface
-  static void ReadThreadHandler(CEXIETHERNET* self);
-  bool Activate();
-  void Deactivate();
-  bool IsActivated();
-  bool SendFrame(const u8* frame, u32 size);
-  bool RecvInit();
-  void RecvStart();
-  void RecvStop();
+  class PhysicalNetworkInterface
+  {
+  protected:
+    CEXIETHERNET* ethRef = nullptr;
+
+  public:
+    virtual bool Activate() { return false; };
+    virtual void Deactivate(){};
+    virtual bool IsActivated() { return false; };
+    virtual bool SendFrame(const u8* frame, u32 size) { return false; };
+    virtual bool RecvInit() { return false; };
+    virtual void RecvStart(){};
+    virtual void RecvStop(){};
+  };
 
-  std::unique_ptr<u8[]> mRecvBuffer;
-  u32 mRecvBufferLength = 0;
+  class TAPPhysicalNetworkInterface : public PhysicalNetworkInterface
+  {
+  public:
+    TAPPhysicalNetworkInterface(CEXIETHERNET* ethRef)
```

Comment:
```suggestion
    explicit TAPPhysicalNetworkInterface(CEXIETHERNET* eth_ref) : PhysicalNetworkInterface(eth_ref) {}
```

with the previous suggestion of supplying a protected constructor.